src/System.Linq/tests/DefaultIfEmptyTests.cs

```diff
@@ -25,86 +24,65 @@ public void SameResultsRepeatCallsEmptyQuery()
         {
             var q = from x in NumberRangeGuaranteedNotCollectionType(0, 0)
                     select x;
-            
+
             Assert.Equal(q.DefaultIfEmpty(88), q.DefaultIfEmpty(88));
 
         }
 
-        [Fact]
-        public void EmptyNullableSourceNoDefaultPassed()
+        public static IEnumerable<object[]> DefaultIfEmpty_TestData()
         {
-            int?[] source = { };
-            int?[] expected = { default(int?) };
+            yield return new object[] { new int[0], 0, new int[] { 0 } };
+            yield return new object[] { new int[] { 3 }, 0, new int[] { 3 } };
+            yield return new object[] { new int[] { 3, -1, 0, 10, 15 }, 0, new int[] { 3, -1, 0, 10, 15 } };
 
-            Assert.Equal(expected, source.DefaultIfEmpty());
+            yield return new object[] { new int[0], -10, new int[] { -10 } };
+            yield return new object[] { new int[] { 3 }, 9, new int[] { 3 } };
+            yield return new object[] { new int[] { 3, -1, 0, 10, 15 }, 9, new int[] { 3, -1, 0, 10, 15 } };
+            yield return new object[] { Enumerable.Empty<int>(), 0, new int[] { 0 } };
         }
 
-        [Fact]
-        public void EmptyNonNullableSourceNoDefaultPassed()
+        [Theory]
+        [MemberData(nameof(DefaultIfEmpty_TestData))]
+        public static void DefaultIfEmpty(IEnumerable<int> source, int defaultValue, int[] expected)
         {
-            int[] source = { };
-            int[] expected = { default(int) };
-
-            Assert.Equal(expected, source.DefaultIfEmpty());
+            IEnumerable<int> result;
+            if (defaultValue == 0)
+            {
+                result = source.DefaultIfEmpty();
+                Assert.Equal(result, result);
+                Assert.Equal(expected, result);
+                Assert.Equal(expected.Length, result.Count());
+                Assert.Equal(expected, result.ToList());
+                Assert.Equal(expected, result.ToArray());
```

Comment:
Or, thinking about it more, is this a test of special implementations in the enumerable returned from DefaultIfEmpty?  If so, makes sense.